woocommerce/woocommerce-android
# Integrate `Automattic-Tracks-Android 2.0.0`

Closes: #2455 
Closes: #3883 

### Context
This PR integrates `Tracks` in version `2.0.0` and adds all changes to comply with updated API.

### How to test
1. Checkout this PR
2. Make sure there's a valid `wc.sentry.dsn` value in `gradle.properties`
3. Apply git's `patch` that is available at the bottom of this PR description
4. Run the app - please log in if you're not
5. Go to Sentry dashboard
6. Go to `Issues` 

Make sure that there's
- [x] `Throwable` error with `This is just a test` message below
7. Click on the `Throwable` error

Make sure that there's
- [x] `site_id` in `tags` section
- [x] `site_url` in `tags` section
- [x] `test` in `tags` section
- [x] `locale` in `tags` section
- [x] `Test message` in `message` section
- [x] `MainActivity` `created` and then `started` in `breadcrumbs` section
- [x] `Test breadcrumb` in `breadcrumbs` section
- [x] User's email, id and username in `user section`

**Make sure to remove this event (by clicking on bin icon) after test**


```
Index: WooCommerce/src/main/kotlin/com/woocommerce/android/util/crashlogging/WCCrashLoggingDataProvider.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- WooCommerce/src/main/kotlin/com/woocommerce/android/util/crashlogging/WCCrashLoggingDataProvider.kt	(revision d7833a7e3c6fbda92bd380fd31a88dfac21b577e)
+++ WooCommerce/src/main/kotlin/com/woocommerce/android/util/crashlogging/WCCrashLoggingDataProvider.kt	(date 1619469640000)
@@ -43,7 +43,7 @@
     }
 
     override fun crashLoggingEnabled(): Boolean {
-        return appPrefs.isCrashReportingEnabled()
+        return true
     }
 
     override fun extraKnownKeys(): List<ExtraKnownKey> {
Index: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreFragment.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreFragment.kt	(revision d7833a7e3c6fbda92bd380fd31a88dfac21b577e)
+++ WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreFragment.kt	(date 1619470055000)
@@ -181,6 +181,13 @@
         tabLayout.addOnTabSelectedListener(tabSelectedListener)
 
         refreshMyStoreStats(forced = this.isRefreshPending)
+
+        crashLogging.recordEvent("Test breadcrumb")
+        crashLogging.sendReport(
+            Throwable(message = "This is just a test"),
+            message = "Test message",
+            tags = mapOf("test" to "tag")
+        )
     }
 
     override fun onResume() {
```
